opm/grid/GraphOfGridWrappers.cpp

```diff
@@ -517,20 +517,21 @@ zoltanPartitioningWithGraphOfGrid(const Dune::CpGrid& grid,
                                   const double zoltanImbalanceTol,
                                   const std::map<std::string, std::string>& params)
 {
-    int rc = ZOLTAN_OK - 1;
     float ver = 0;
     struct Zoltan_Struct *zz;
     int changes, numGidEntries, numLidEntries, numImport, numExport;
     ZOLTAN_ID_PTR importGlobalGids, importLocalGids, exportGlobalGids, exportLocalGids;
     int *importProcs, *importToPart, *exportProcs, *exportToPart;
     int argc=0;
     char** argv = 0 ;
-    rc = Zoltan_Initialize(argc, argv, &ver);
-    zz = Zoltan_Create(cc);
-    if ( rc != ZOLTAN_OK )
-    {
+    int rc = Zoltan_Initialize(argc, argv, &ver);
+    if (rc != ZOLTAN_OK) {
         OPM_THROW(std::runtime_error, "Could not initialize Zoltan!");
     }
+    zz = Zoltan_Create(cc);
+    if (rc != ZOLTAN_OK) {
+        OPM_THROW(std::runtime_error, "Could not create Zoltan data structures!");
+    }
```

Comment:
The `rc` here is the one from the `Zoltan_Initialize()` call above, at least if there's no global reference anywhere, so I don't see how it could be anything other than `ZOLTAN_OK` in this instance.  Maybe you wanted `zz == nullptr` instead?

Reply:
yes! thanks, that was stupid half-work.